### ISSUE TITLE
Labeling of amenity=parcel_locker incompatible with the goal of constructive mapper feedback

### ISSUE THREAD

**Issue by imagico** [gravitystorm/openstreetmap-carto#4629]
As a result of #4512 we now have a labeling logic for `amenity=parcel_locker` of the form:

https://github.com/gravitystorm/openstreetmap-carto/blob/9bda286762d8a494a7337df6ee10cb14b0f637fa/project.mml#L1462

Technically this renders as a label the brand (see https://github.com/gravitystorm/openstreetmap-carto/issues/4575#issuecomment-1158651857 for some comments on the specific semantic issues of that tag), the operator or the name - whichever is tagged, in that order of preference, plus the ref tag, separated by a newline if there is a (brand or operator or name).

As already explained in the PR (https://github.com/gravitystorm/openstreetmap-carto/pull/4512#issuecomment-1100308448) this renders features tagged `amenity=parcel_locker` + `name=foo` with a _foo_ label while the name tag of those features practically never contains an actual name and almost always contains a compound labeling string subjectively designed to the individual mappers liking.  But it is even worse than just plainly rendering the name tag, it essentially communicates to the mapper: You can tag your subjectively desired labeling string in either `name`, `brand` or `operator` - it does not matter where, it is all the same.  So it essentially actively sabotages attempts at supporting consistent use of tags.

I was not sure if i should open this issue considering  i already brought this matter up in #4512 and this was merged without addressing my concern (which is fine).  But it is likely that with the lot of noise on that PR other maintainers were not consciously aware of this problem.  So what i essentially want to find out with this issue is if we still have consensus on what for the whole history of this style has been a core guiding principle - to provide constructive mapper feedback, to support mappers in consistent use of tags and to be intuitively understandable for mappers in how a certain tagging results in the rendering we produce.  If we still have consensus about this goal this labeling logic can't be acceptable and if we accept this labeling logic we clearly abandon this goal.

**Comment by pnorman** [comment]
I don't agree that there is a problem with using multiple tags to derive the label, but would consider a PR changing it to just render the name.

Would you only render brand, plus maybe ref?

**Comment by pnorman** [comment]
> * do you disagree that coalescing brand, operator and name communicates to the mapper:  You can tag your subjectively desired labeling string in either name, brand or operator - it does not matter where, it is all the same?
> * do you disagree with the assessment that the labeling logic used here is not intuitively understandable for the mapper, especially since practically the tags brand, operator, name and ref will often have overlapping content?

I do not find anything with coalescing that conflicts with general purpose of the style. In particular, I find it neutral on "it's an important feedback mechanism for mappers to validate their edits and helps to prevent unfavorable fragmentation of tag use" and positive on the other three purposes.

> * do you disagree with the assessment that the name tag is almost never used on `amenity=parcel_locker` to tag an actual name?

I'm not sure what you mean by an actual name. 

> Note the latter is of course not a problem unique to amenity=parcel_locker, we also have this for addresses:

I don't consider it a problem for addresses either.

I'm not a huge fan of the technical complexity in using labels that aren't a single database column, but that's a technical objection based around adding complexity to SQL, MSS, and reworking indexes.

**Comment by imagico** [comment]
> I do not find anything with coalescing that conflicts with general purpose of the style. In particular, I find it neutral on "it's an important feedback mechanism for mappers to validate their edits and helps to prevent unfavorable fragmentation of tag use" and positive on the other three purposes.

That is not what i asked.

I find it pretty obvious that the labeling logic used here is non-intuitive for the mapper (meaning the vast majority of mappers will not develop a correct understanding of how the label derives from the tagging merely by studying the map) and can lead to various irritating and misleading effects when a mapper changes tagging and is therefore counterproductive in terms of mapper feedback.

To give an example:

You have a node with `amenity=parcel_locker` and `name=DHL Packstation 120` and a mapper adds `brand=DHL Packstation` and `ref=120`.  The label does not change:

![parcel locker](https://imagico.de/files/parcel_locker_label.png)

but the label is sourced from different tags in both cases.

> I'm not sure what you mean by an actual name.

I believe the technical term here in English is *proper name*, as in

https://en.wikipedia.org/wiki/Proper_name_(philosophy)

The name tag by broad consensus and in practical use in general is predominantly used to record proper names.  However on amenity=parcel_locker it almost never is.  If you disagree i would ask you to point to examples where the name tag of an amenity=parcel_locker contains a proper name.

> I'm not a huge fan of the technical complexity in using labels that aren't a single database column, but that's a technical objection based around adding complexity to SQL, MSS, and reworking indexes.

My concern has nothing to do with technical complexity, it is about the social function the style has in the OSM community.

**Comment by Lukas458** [comment]
> The name tag by broad consensus and in practical use in general is predominantly used to record proper names. However on amenity=parcel_locker it almost never is.

**Comment by Lukas458** [comment]
> I find it pretty obvious that the labeling logic used here is non-intuitive for the mapper

This is true. But maybe, just concentrating on both the Tags brand and ref to render could make more sense here than concentrating on putting out the parcel locker's text out of one single key-value-pair so that this could be (intuitively) then the name only.

So as you said, dropping the rendering of name=* at parcel lockers completely could give mappers the feedback to user others Tags, Brand and ref. Of course these are two Tags here and not just one like for ameniy=atm with its operator.

But See, it's the same at Mountain peaks for example. For having the height rendered, there's the tag ele=* which is used together with name.

The only special thing is in this case, that the "name rendered" is constructed with two Tags which are *both not in the name-tag,* but in other tags.

Rendering brand+ref and not name prevents misusing the name tag. For amenity=atm and amenity=vending_machine operators for example we did similar things for it.

**Comment by imagico** [comment]
> In my eyes, it's the same at mountain peaks for example. For having the height rendered, there's the tag ele=* which is used together with name.

Note for mountain peaks we match the ele tag against a regular expression:

https://github.com/gravitystorm/openstreetmap-carto/blob/4b58c59093f3973566ad8f0ee679560e8c6c8d75/project.mml#L1468

to make sure we only render ele values which are numbers (and we round them to integer meters).

Rendering _brand + ref_ or _operator + ref_ would quite definitely be an improvement over the status quo.  I don't think either of these is ideal though without a differentiation in styling that shows which part of the label is which tag.

We have one other case (in addition to elevations, addresses and `amenity=parcel_locker`) where we compose a label from multiple tags - that is motorway junctions:

https://github.com/gravitystorm/openstreetmap-carto/blob/4b58c59093f3973566ad8f0ee679560e8c6c8d75/style/roads.mss#L2757-L2764

There are two important things that makes this much less prone to confusion:

* at z<13 only _ref_ is rendered thereby giving visual feedback on that tag in isolation
* it uses `text-wrap-character: ";"` - so there are usually no line breaks in the label apart from the one separating _name_ and _ref_ thereby making the labeling logic much more intuitive

**Comment by RicoElectrico** [comment]
My arguments for leaving as-is:
- I think you overstate mapper feedback in this particular case. What drives tagging is iD presets. Which, incidentally, does have separate fields for brand and ref (and also uses brand suggestion index!).  Only when something is not displayed or properly differentiated when necessary (like unpaved roads), then some mappers start fiddling with it.
- If you drop rendering ref=* altogether it'll cause people to tag for the renderer anyway.
- Each major parcel locker network shares their locations, and with that I expect that some bored mappers will emerge to maintain networks in their country by the means of overpass queries and cross-checks with said external datasets.

**Comment by Lukas458** [comment]
An option to remove/drop the rendering of a name=* (if tagged) on parcel_lockers would it maybe be, when the name suggestion index (whose presets are used for parcel lockers in most cases) would have removed all "name" tags in[ its parcel locker presets](https://github.com/osmlab/name-suggestion-index/blob/main/data/brands/amenity/parcel_locker.json) , but unfurtunately, this isn't the case at the moment even though the "names" there aren't those and in all cases it is either completely the same as the value for "brand" or its a combination of "brand" + the word "parcel locker" and/or "ref" (here, however, there is also far from agreement, with one NSI writing that one in the name, with the other that...))

**Comment by Lukas458** [comment]
The problem is also, that if the NSI has a "name" in a parcel_loker preset and the preset for this parcel locker is in use, but then comes a fitting parcel locker where the name isn't tagged, the NSI shows this parcel locker als tagged imcompletely.

I have seen so many changesets where just the suggestions of NSI were implemented on those amenities without any other changes or something, just to satisfy NSI. So in bottom line, this will lead to that the parcel lockers of specific brands will **all** get tagged with a name.

But that is not so much a concrete decision in individual cases or a decision made by the local community, where a name is being considered, but the NSI simply leads to it. I know I should open an issue at the NSI for this.

**Comment by Lukas458** [comment]
> a differentiation in styling that shows which part of the label is which tag

interesting thought!
